sdk-lib/src/androidTest/java/cash/z/ecc/android/sdk/db/DatabaseCoordinatorTest.kt

```diff
@@ -0,0 +1,179 @@
+package cash.z.ecc.android.sdk.db
+
+import androidx.test.filters.SmallTest
+import cash.z.ecc.android.sdk.test.getAppContext
+import cash.z.ecc.android.sdk.type.ZcashNetwork
+import cash.z.ecc.fixture.DatabaseNameFixture
+import cash.z.ecc.fixture.DatabasePathFixture
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+
+class DatabaseCoordinatorTest {
+
+    private val dbCoordinator = DatabaseCoordinator.getInstance(getAppContext())
+
+    @Before
+    fun clear_test_files() {
+        val databaseDir = DatabasePathFixture.new(baseFolderPath = DatabasePathFixture.DATABASE_DIR_PATH)
+        val noBackupDir = DatabasePathFixture.new(baseFolderPath = DatabasePathFixture.NO_BACKUP_DIR_PATH)
+        File(databaseDir).deleteRecursively()
+        File(noBackupDir).deleteRecursively()
+    }
+
+    // Sanity check of the database coordinator instance and its thread-safe implementation. Our aim
+    // here is to run two jobs in parallel at the same time to test mutex implementation and correct
+    // DatabaseCoordinator function call result.
+    @Test
+    @SmallTest
+    fun mutex_test() = runTest {
```

Comment:
To fully exercise this, we'd probably want to loop many times.

When testing potential race conditions in a ContentProvider implementation, I did something like this:

https://github.com/twofortyfouram/android-monorepo/blob/4023d30d5b44b2fd15206c3435829e999e7bbd02/mementoImplLib/src/androidTest/java/com/twofortyfouram/memento/provider/MementoContentProviderTest.java#L1500

Reply:
Yep, sounds good. I've implemented a new additional stress test, which runs the original one sequentially 50 times to test, if the original test can be flaky or not. And it looks good for now.